wiv.js

```diff
@@ -151,54 +151,89 @@ function wiv(params) {
     ctx.beginPath();
     ctx.clearRect(0, 0, canvas.width, canvas.height);
 
-    let x = height * 2 + thickness
-    let y = height - Math.sin(((x - frame) * tightness) * Math.PI / 180) * height + thickness;
-
-    //range of the sin function will be [-1 -> 1] * height. Since the logic will never want negative values for y (or clipping), it must have a vertical offset that takes all parameters into account
+    // range of the sin function will be [-1 -> 1] * height. Since the logic will never want negative values for y (or clipping), it must have a vertical offset that takes all parameters into account
     let offset = height + Math.max(thickness, imageMode ? imageSize : 0)
 
-    //draw top
-    for (x = height * 3; x <= canvas.width - (height * 3); x += increment) {
-      y =  offset + (Math.sin(((x - frame) * tightness) * Math.PI / 180) * height);
+    function calculateTopYValue(x) {
+      return offset + (Math.sin(((x - frame) * tightness) * Math.PI / 180) * height);
+    }
+
+    function calculateRightXValue(y) {
+      return (canvas.width - offset) - (Math.cos(((y - frame) * tightness) * Math.PI / 180) * height);
+    }
+
+    function calculateBottomYValue(x) {
+      return (canvas.height - offset) + (Math.sin(((x - frame) * tightness) * Math.PI / 180) * height);
+    }
+
+    function calculateLeftXValue(y) {
+      return offset + (Math.cos(((y - frame) * tightness) * Math.PI / 180) * height);
+    }
+
+    function findIntersection(start, end, firstFunc, secondFunc, trim = 2) {
+      let reverse = (end < start);
+      let coordList = [];
+      for (let x = start; (reverse && x >= end) || (!reverse && x <= end); x += reverse ? -1 : 1) {
+        let y = firstFunc(x);
+        let newX = secondFunc(y);
+        coordList.unshift({
+          x: x,
+          y: y
+        });
+        coordList = coordList.slice(0, trim);
+        if ((reverse && newX >= x) || (!reverse && newX <= x)) {
+          return coordList.pop();
+        }
+      }
+      return coordList.pop();
+    }
+
+    let topLeft = findIntersection(height * 4, 0, calculateTopYValue, calculateLeftXValue);
```

Comment:
can we move this to a constant. Its the length of the side - the corner bounding boxes?